qemu/tests/cpu_L3_cache.py

```diff
@@ -0,0 +1,56 @@
+import re
+import logging
+
+from avocado.utils import process
+
+from virttest import env_process
+from virttest import error_context
+from virttest import utils_misc
+from virttest import utils_qemu
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Check L3 cache present to guest
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+
+    def boot_and_check_guest(machine_type, check_L3=False):
+        """
+        Boot guest and check L3 cache inside guest
+
+        :param machine_type: Boot guest with which machine type
+        :param check_L3: if L3 cache should exist on guest
+        """
+        params['machine_type'] = machine_type
+        params['start_vm'] = 'yes'
+        vm_name = params['main_vm']
+        L3_existence = 'present' if check_L3 else 'not present'
+        logging.info('Boot guest with machine type %s and expect L3 cache %s'
+                     ' inside guest' % (machine_type, L3_existence))
+        env_process.preprocess_vm(test, params, env, vm_name)
+        vm = env.get_vm(vm_name)
+        session = vm.wait_for_login()
+        output = session.cmd_output('lscpu')
+        session.close()
+        vm.destroy()
+        L3_present = 'L3 cache' in output
+        if check_L3 ^ L3_present:
+            test.fail('L3 cache should %s inside guest for machine type %s'
+                      ' while it is not' % (L3_existence, machine_type))
+
+    qemu_bin = utils_misc.get_qemu_binary(params)
+    latest_machines = process.getoutput('%s -machine ?| grep alias' %
+                                        qemu_bin, shell=True).split('\n')
```

Comment:
we will get the two latest machine_type in here, both `pc_machine_type` and `q35_machine_type` will be booted while we test the case with default machine type `pc`.

Reply:
@nanliu-r In the test case it says 'pc-i440fx-rhel7.4.0 or pc-q35-rhel7.4.0' so I test both of them, so here you mean we don't need to test q35 so it should be removed, right?

Reply:
@nanliu-r Or you mean we should test with `pc` instead of the latest one I get here? If it is what you mean, then I think current approach is okay, cause `pc` always alias to the latest one, that's why I get it this way. 
And, actually that's for multi-arch support previously, although now it is not needed anymore, but I think it might be more precise, having said that, if you insist hard coding `pc` or `q35`, I could modify it. Let me know your thought, thanks.
```
# /usr/libexec/qemu-kvm -machine ?
Supported machines are:
pc                   RHEL 7.6.0 PC (i440FX + PIIX, 1996) (alias of pc-i440fx-rhel7.6.0)
pc-i440fx-rhel7.6.0  RHEL 7.6.0 PC (i440FX + PIIX, 1996) (default)
pc-i440fx-rhel7.5.0  RHEL 7.5.0 PC (i440FX + PIIX, 1996)
pc-i440fx-rhel7.4.0  RHEL 7.4.0 PC (i440FX + PIIX, 1996)
pc-i440fx-rhel7.3.0  RHEL 7.3.0 PC (i440FX + PIIX, 1996)
pc-i440fx-rhel7.2.0  RHEL 7.2.0 PC (i440FX + PIIX, 1996)
pc-i440fx-rhel7.1.0  RHEL 7.1.0 PC (i440FX + PIIX, 1996)
pc-i440fx-rhel7.0.0  RHEL 7.0.0 PC (i440FX + PIIX, 1996)
q35                  RHEL-8.2.0 PC (Q35 + ICH9, 2009) (alias of pc-q35-rhel8.2.0)
pc-q35-rhel8.2.0     RHEL-8.2.0 PC (Q35 + ICH9, 2009)
pc-q35-rhel8.1.0     RHEL-8.1.0 PC (Q35 + ICH9, 2009)
pc-q35-rhel8.0.0     RHEL-8.0.0 PC (Q35 + ICH9, 2009)
pc-q35-rhel7.6.0     RHEL-7.6.0 PC (Q35 + ICH9, 2009)
pc-q35-rhel7.5.0     RHEL-7.5.0 PC (Q35 + ICH9, 2009)
pc-q35-rhel7.4.0     RHEL-7.4.0 PC (Q35 + ICH9, 2009)
pc-q35-rhel7.3.0     RHEL-7.3.0 PC (Q35 + ICH9, 2009)
```

Reply:
@vivianQizhu I mean we should separate the step according to the automation parameter "`machine_type`", but since the current code is easier it is ok for me, please add more steps describes clarifying we will cover all scenario once, thanks.